src/main/kotlin/no/nav/medlemskap/domene/Periode.kt

```diff
@@ -0,0 +1,53 @@
+package no.nav.medlemskap.domene
+
+import no.nav.medlemskap.regler.common.startOfDayInstant
+import org.threeten.extra.Interval
+import java.time.Instant
+import java.time.LocalDate
+
+data class Periode(
+    val fom: LocalDate?,
+    val tom: LocalDate?
+) {
+    fun erGyldigPeriode(): Boolean {
+        return (fomNotNull().isBefore(tomNotNull()) || fomNotNull().isEqual(tomNotNull()))
+    }
+
+    fun overlapper(annenPeriode: Periode): Boolean {
+        if (!erGyldigPeriode()) {
+            return false
+        }
+
+        return interval().overlaps(annenPeriode.interval()) || interval().encloses(annenPeriode.interval())
```

Comment:
Er det noe vits med sjekk på om intervall encloses? Vil ikke overlapp sjekken alltid være true dersom encloses er true?

Reply:
Bra spørsmål, skal sjekke det litt nærmere. Dette er egentlig bare noe jeg har videreført fra sånn som det var før.

Reply:
Jeg fant ut at det ikke er nødvendig å sjekke på encloses når man sjekker for overlapp